From 5b66c9436c45bed5e9967003f8de4425cbcc8e2f Mon Sep 17 00:00:00 2001
From: Yan Markman <ymarkman@marvell.com>
Date: Thu, 8 Oct 2026 15:54:44 +0300
Subject: [PATCH 1/2] platform marvell-prestera: add db98dx73xx_cn913 AC5X_DB

SONiC main repo:
  platform/marvell-prestera

Add new platform-device db98dx73xx_cn913 (AC5X_DB)

Signed-off-by: Yan Markman <ymarkman@marvell.com>
---
 platform/marvell-prestera/one-image.mk        | 1 +
 platform/marvell-prestera/platform-marvell.mk | 5 +++++
 platform/marvell-prestera/platform_arm64.conf | 4 ++++
 3 files changed, 10 insertions(+)

diff --git a/platform/marvell-prestera/one-image.mk b/platform/marvell-prestera/one-image.mk
index 1a3a5216e..816992940 100644
--- a/platform/marvell-prestera/one-image.mk
+++ b/platform/marvell-prestera/one-image.mk
@@ -13,6 +13,7 @@ $(SONIC_ONE_IMAGE)_INSTALLS += $(MRVL_PRESTERA_DEB)
 $(SONIC_ONE_IMAGE)_LAZY_INSTALLS += $(NOKIA_7215_PLATFORM) \
 				$(AC5X_RD98DX35xx_PLATFORM) \
 				$(AC5X_RD98DX35xxCN9131_PLATFORM) \
+				$(AC5X_DB98DX73xxCN9131_PLATFORM) \
 				$(AC5P_RD98DX45xxCN9131_PLATFORM)
 else ifeq ($(CONFIGURED_ARCH),armhf)
 $(SONIC_ONE_IMAGE)_INSTALLS += $(MRVL_PRESTERA_DEB)
diff --git a/platform/marvell-prestera/platform-marvell.mk b/platform/marvell-prestera/platform-marvell.mk
index c550491e3..d2934063a 100644
--- a/platform/marvell-prestera/platform-marvell.mk
+++ b/platform/marvell-prestera/platform-marvell.mk
@@ -2,6 +2,7 @@
 
 AC5X_RD98DX35xx_PLATFORM_MODULE_VERSION = 1.0
 AC5X_RD98DX35xxCN9131_PLATFORM_MODULE_VERSION = 1.0
+AC5X_DB98DX73xxCN9131_PLATFORM_MODULE_VERSION = 1.0
 AC5P_RD98DX45xxCN9131_PLATFORM_MODULE_VERSION = 1.0
 FALCON_PLATFORM_MODULE_VERSION = 1.0
 
@@ -17,6 +18,10 @@ AC5X_RD98DX35xxCN9131_PLATFORM = sonic-platform-rd98dx35xx-cn9131_$(AC5X_RD98DX3
 $(AC5X_RD98DX35xxCN9131_PLATFORM)_PLATFORM = arm64-marvell_rd98DX35xx_cn9131-r0
 $(eval $(call add_extra_package,$(AC5X_RD98DX35xx_PLATFORM),$(AC5X_RD98DX35xxCN9131_PLATFORM)))
 
+AC5X_DB98DX73xxCN9131_PLATFORM = sonic-platform-db98dx73xx-cn9131_$(AC5X_DB98DX73xxCN9131_PLATFORM_MODULE_VERSION)_$(CONFIGURED_ARCH).deb
+$(AC5X_DB98DX73xxCN9131_PLATFORM)_PLATFORM = arm64-marvell_db98DX73xx_cn9131-r0
+$(eval $(call add_extra_package,$(AC5X_RD98DX35xx_PLATFORM),$(AC5X_DB98DX73xxCN9131_PLATFORM)))
+
 AC5P_RD98DX45xxCN9131_PLATFORM = sonic-platform-rd98dx45xx-cn9131_$(AC5P_RD98DX45xxCN9131_PLATFORM_MODULE_VERSION)_$(CONFIGURED_ARCH).deb
 $(AC5P_RD98DX45xxCN9131_PLATFORM)_PLATFORM = arm64-marvell_rd98DX45xx_cn9131-r0
 $(eval $(call add_extra_package,$(AC5X_RD98DX35xx_PLATFORM),$(AC5P_RD98DX45xxCN9131_PLATFORM)))
diff --git a/platform/marvell-prestera/platform_arm64.conf b/platform/marvell-prestera/platform_arm64.conf
index 5c4be072c..55174fe64 100644
--- a/platform/marvell-prestera/platform_arm64.conf
+++ b/platform/marvell-prestera/platform_arm64.conf
@@ -44,6 +44,10 @@ case $PLATFORM in
         mmc_bus="mmc0:0001";
         fdt_fname="/boot/cn9131-db-comexpress.dtb";
         fit_conf_name="#conf_cn9131";;
+    arm64-marvell_db98DX73xx_cn9131-r0) PLATFORM_CN9131=1;
+        mmc_bus="mmc0:0001";
+        fdt_fname="/boot/cn9131-db-comexpress.dtb";
+        fit_conf_name="#conf_cn9131";;
     arm64-marvell_rd98DX45xx_cn9131-r0) PLATFORM_CN9131=1;
         scsi_bus="1:0:0:0";
         if [ "$install_env" != "onie" ]; then
-- 
2.34.1

